### PR TITLE
Add /rhp/broadcast/:id endpoint to worker

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
Adds a new endpoint to the worker which allows for submitting a contract's revision to the network.
This can serve multiple purposes such as periodically publishing revisions to limit a host's ability to submit an old revision in case they lose data.

### CODE REVIEW

File: worker/worker.go [hunk 1]
```diff
@@ -1037,6 +1090,7 @@ func (w *worker) Handler() http.Handler {
 		"GET    /account/:hostkey": w.accountHandlerGET,
 		"GET    /id":               w.idHandlerGET,
 
+		"POST   /rhp/broadcast/:id":   w.rhpBroadcastHandler,
```
Comment (peterjan) [inline worker/worker.go:1093]: I think `/rhp/contract/:id/broadcast` is better

File: worker/worker.go [hunk 2]
```diff
@@ -517,6 +519,57 @@ func (w *worker) rhpFormHandler(jc jape.Context) {
...
+	c, err := w.bus.Contract(ctx, fcid)
+	if jc.Check("could not get contract", err) != nil {
+		return
+	}
+	gp, err := w.bus.GougingParams(ctx)
+	if jc.Check("could not get gouging parameters", err) != nil {
+		return
+	}
+	rk := w.deriveRenterKey(c.HostKey)
+	ctx = WithGougingChecker(ctx, w.bus, gp)
```
Comment (peterjan) [inline worker/worker.go:539]: Do you need the gouging checker on the `ctx` here? Seems like you are not using it

File: worker/worker.go (hunk 1 above)
Comment (ChrisSchinnerl) [reply worker/worker.go:1093]: I feel like `rhp` would be a bit much if we also want to scope it to contracts. Maybe `/contract/:id/broadcast`?
The `rhp` segment is not really meaningful imo and if we redid it from scratch I'd probably also scope the contract formation and renewal under `/contract/`.

File: worker/client.go [hunk 3]
```diff
@@ -31,6 +31,12 @@ func (c *Client) ID(ctx context.Context) (id string, err error) {
 	return
 }
 
+// RHPBroadcast broadcasts the latest revision for a contract.
+func (c *Client) RHPBroadcast(ctx context.Context, fcid types.FileContractID) (err error) {
```
Comment (peterjan) [inline worker/client.go:35]: `nit` update the client method?

**Review by peterjan — APPROVED** [review]

**Pull request merged**